src/braket/circuits/circuit.py

```diff
@@ -773,6 +805,56 @@ def apply_initialization_noise(
 
         return apply_noise_to_moments(self, noise, target_qubits, "initialization")
 
+    def set_parameter_values(
+        self, param_values: Dict[str, Number], strict: bool = False
+    ) -> Circuit:
+        """
+        Sets FreeParameters based upon their name and values passed in. If parameters
+        share the same name, all the parameters of that name will be set to the mapped value.
+
+        Args:
+            param_values (Dict[str, Number]):  A mapping of FreeParameter names
+                to a value to assign to them.
+
+            strict (bool, optional): If True, raises a ValueError if none of the FreeParameters
+                in param_values appear in the circuit. False by default."
+
+        Returns:
+            Circuit: Returns a circuit with all present parameters fixed to their respective
+                values.
+
+        Raises:
+            ValueError: If there are no parameters that match the key for the arg
+                param_values.
+        """
+        fixed_circ = Circuit()
+        if strict:
```

Comment:
Apparently I commented on the diff rather than the pull request. Please see my comments in the diff.

Reply:
Ack